config/migrate_test.go

```diff
@@ -1,80 +1,134 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-package config
+package config_test
 
 import (
+	"io/ioutil"
+	"os"
+	"path"
 	"testing"
 
+	"github.com/mattermost/mattermost-server/v5/config"
 	"github.com/mattermost/mattermost-server/v5/model"
-	"github.com/mattermost/mattermost-server/v5/testlib"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
-func getDsn(driver string, source string) string {
-	if driver == model.DATABASE_DRIVER_MYSQL {
-		return driver + "://" + source
+func TestMigrate(t *testing.T) {
+	files := []string{
+		"IdpCertificateFile",
+		"PublicCertificateFile",
+		"PrivateKeyFile",
+		"internal.crt",
+		"internal2.crt",
 	}
-	return source
-}
-
-func TestMigrateDatabaseToFile(t *testing.T) {
-	if testing.Short() {
-		t.SkipNow()
+	filesData := make([]string, len(files))
+	for i := range files {
+		filesData[i] = model.NewId()
 	}
-	helper := testlib.NewMainHelper()
-	sqlSettings := helper.GetSQLSettings()
-	fileDSN := "config.json"
-	files := []string{"IdpCertificateFile", "PublicCertificateFile", "PrivateKeyFile"}
-	data := []byte("aaaaa")
-	ds, err := NewDatabaseStore(getDsn(*sqlSettings.DriverName, *sqlSettings.DataSource))
-	defer ds.Close()
-	defer func() {
-		defaultCfg := &model.Config{}
-		defaultCfg.SetDefaults()
-		ds.Set(defaultCfg)
-	}()
-	require.NoError(t, err)
-	config := ds.Get()
-	config.SamlSettings.IdpCertificateFile = &files[0]
-	config.SamlSettings.PublicCertificateFile = &files[1]
-	config.SamlSettings.PrivateKeyFile = &files[2]
-	_, err = ds.Set(config)
-	require.NoError(t, err)
-
-	for _, file := range files {
-		err = ds.SetFile(file, data)
+
+	setup := func(t *testing.T) {
+		os.Clearenv()
+		t.Helper()
```

Comment:
This is actually to prevent any developer-overridden `MM_*` settings from influencing the tests, since the `config` package consumes settings like `MM_SERVICESETTINGS_SITEURL=http://example.com` and such.